packages/babel-preset-minify/__tests__/minify-env-tests.js

```diff
@@ -196,4 +196,59 @@ describe("preset along with env", () => {
       console.log(foo);
     `
   );
+
+  thePlugin(
+    "should fix issue#829 mangling after function name",
+    `
+      function foo() {
+        let con = console;
+
+        return {
+          a(bar) {
+            con.log(bar);
+          }
+        };
+      }
+    `,
+    `
+      function foo() {
+        var b = console;
+        return {
+          a: function d(c) {
```

Comment:
Shouldn’t this be `function a` so that its `.name` is correct?

Reply:
Handled by `keepFnName` option.

Reply:
yes. Correct. Deadcode should have taken care of that (have to debug why it didn't in this case), and we come back to plugin ordering problem - https://github.com/babel/minify/issues/422

Reply:
I guess it's the same issue as this one - we have to add the same thing in DCE (register this binding before dce runs).